### PR TITLE
chore: cleanup unused flow

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

We had some legacy flow for onboarding via post page (this got changed to uniform onboarding page)
However some elements still referred to this flow although it should't exist.

The only edge-case was the `registerLocalFilters` function, which was called on `onboarding` page if user didn't select tags. It's technically impossible, but saw in clickhouse there's about 3 a week that have this. (Not sure how)

Removing it doesn't break anything though as their flow still works.

Related discussion:
https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1727951097888609

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-631 #done 

### Preview domain
https://as-631-cleanup-unused-code.preview.app.daily.dev

### CODE REVIEW

**Review by omBratteng — APPROVED** [review]

File: packages/shared/src/components/modals/ArticlePostModal.tsx [hunk 1]
```diff
@@ -24,11 +23,11 @@ export default function ArticlePostModal({
   onRemovePost,
   ...props
 }: ArticlePostModalProps): ReactElement {
-  const { showArticleOnboarding } = useContext(OnboardingContext);
   const { position, onLoad } = usePostNavigationPosition({
     isDisplayed: props.isOpen,
-    offset: showArticleOnboarding ? ONBOARDING_OFFSET : 0,
+    offset: 0,
   });
+  console.log('position', position);
```
Comment (rebelchris) [inline packages/shared/src/components/modals/ArticlePostModal.tsx:30]:
```suggestion
```

File: packages/storybook/stories/components/HorizontalScroll.stories.tsx [hunk 2]
```diff
@@ -2,7 +2,6 @@ import type { Meta, StoryObj } from '@storybook/react';
 import HorizontalScroll from '@dailydotdev/shared/src/components/HorizontalScroll/HorizontalScroll';
 import React, {
   ReactElement,
-  Ref,
```
Comment (rebelchris) [inline packages/storybook/stories/components/HorizontalScroll.stories.tsx:5]: unused

File: packages/storybook/stories/extension/ShortcutLinks.stories.tsx [hunk 3]
```diff
@@ -1,11 +1,5 @@
 import type { Meta, StoryObj } from '@storybook/react';
-import { HttpResponse, http } from 'msw';
```
Comment (rebelchris) [inline packages/storybook/stories/extension/ShortcutLinks.stories.tsx:2]: All unused

File: packages/shared/src/components/Feed.spec.tsx [hunk 4]
```diff
@@ -232,23 +231,13 @@ const renderComponent = (
...
-            <Feed
-              feedQueryKey={['feed']}
-              feedName={feedName}
-              query={ANONYMOUS_FEED_QUERY}
-              variables={variables}
-            />
-          </OnboardingContext.Provider>
+          <Toast autoDismissNotifications={false} />
+          <Feed
+            feedQueryKey={['feed']}
```
Comment (capJavert) [inline packages/shared/src/components/Feed.spec.tsx:236]: We have OtherFeed or some constant for this?

**Review by capJavert — APPROVED** [review]

File: packages/shared/src/components/Feed.spec.tsx (hunk 4 above)
Comment (rebelchris) [reply packages/shared/src/components/Feed.spec.tsx:236]: The key you mean?
Probably didn't really change anything here 😅

Comment (capJavert) [reply packages/shared/src/components/Feed.spec.tsx:236]: non blocking just checking if we have constant in the meantime 👍

Comment (rebelchris) [reply packages/shared/src/components/Feed.spec.tsx:236]: I actually checked, but we don't specific that one somewhere at the moment as enum.

**Pull request merged**